### DISCUSSION TITLE
koa-body and [@types/koa-bodyparser] type incompatibility.

### DISCUSSION THREAD

**Discussion by maslade** [DefinitelyTyped/DefinitelyTyped#62237] [Issues with a @types package]
There has been a history of disagreement between these two packages about the type of `Request.body` and it looks like [this change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61549) has resurfaced it.

The type in the latest versions of `koa-body` (5.0, 4.2) is ` any`. The above linked PR changes this to `Record<string, unknown>;`. I'm not familiar enough with the ecosystem to know which package should be the authority, but it seems like `koa-body`. Using the latest from both packages (`@types/koa-bodyparser` and `koa-body`) I get the error:

```sh
#17 221.7 node_modules/@types/koa-bodyparser/index.d.ts(25,9): error TS2717: Subsequent property declarations must have the same type.  Property 'body' must be of type 'any', but here has type 'Record<string, unknown> | undefined'.
```

Some relevant historical threads:
* https://github.com/koajs/koa-body/pull/119 ([related issue](https://github.com/koajs/koa-body/issues/118))
* https://github.com/DefinitelyTyped/DefinitelyTyped/commit/110ceb19ebb7186cd5439f5d15a68b4976b71c93#diff-2ab9c657953d22d2246f6bef3278b6c17dc9801ed3c0a5964eadb5f974325645R25

[a comment by an automated account is not shown]

**Comment by pkuczynski** [comment]
`koa-body` should use `Record<string, unknown> | undefined`, as `@types/koa-bodyparser` correctly highlights that `any` defeats the purpose of strong typing...

**Comment by amaschas** [comment]
I don't have a strong opinion on what the type should be, but this problem is longstanding is actually the reason why I got tagged as an owner on this project. I agree in generally that we should strive to type things correctly, but honestly I see this as more of a Typescript issue, since the actual problem is that Typescript does not offer a mechanism to easily resolve type conflicts in packages. I'd need to know more about why the `Record` type is correct here. From a usability perspective, it seems like a lot of users prefer `any`, and I tend to defer to usability over absolute correctness of typing.

**Comment by Sozialarchiv** [comment]
Until 4.3.7 everything is working fine.
